packages/cli/src/ui/hooks/useKittyKeyboardProtocol.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useState } from 'react';
+import {
+  isKittyProtocolEnabled,
+  isKittyProtocolSupported,
+} from '../utils/kittyProtocolDetector.js';
+
+export interface KittyProtocolStatus {
+  supported: boolean;
+  enabled: boolean;
+  checking: boolean;
+}
+
+/**
+ * Hook that returns the cached Kitty keyboard protocol status.
+ * Detection is done once at app startup to avoid repeated queries.
+ */
+export function useKittyKeyboardProtocol(): KittyProtocolStatus {
+  const [status] = useState<KittyProtocolStatus>({
```

Comment:
what is the purpose of using this as a hook instead of just calling isKittyProtocolSupported and isKittyProtocolEnabled directly from callers that care?